### PR TITLE
Cut version 0.4.0

### PR DESCRIPTION
**Pull request by spacejam**

### CODE REVIEW

File: CRATE.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+[![License Apache 2](https://img.shields.io/badge/License-Apache2-blue.svg)](https://www.apache.org/licenses/LICENSE-2.0)
+[![Crates.io](https://img.shields.io/crates/v/nats.svg)](https://crates.io/crates/nats)
+[![Documentation](https://docs.rs/nats/badge.svg)](https://docs.rs/nats/)
+[![Build Status](https://travis-ci.org/nats-io/nats.rs.svg?branch=master)](http://travis-ci.org/nats-io/nats.rs)
+
+Warning: as of version 0.4.0, this crate has been rewritten, and several public
+APIs have changed. Please see the [docs](https://docs.rs/nats), the [new
+repo](https://github.com/nats-io/nats.rs), and [the
+changelog](https://github.com/nats-io/nats.rs/blob/master/CHANGELOG.md) for
+details.
```
Comment (spacejam) [inline CRATE.md:19]: This is the initial text for the warning we talked about adding to this.

Comment (derekcollison) [reply CRATE.md:19]: I would add a note that the maintainers have changed and this is a new code base supported by Synadia, etc. Would make it a bit more detailed.

File: CRATE.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    A <a href="https://www.rust-lang.org/">Rust</a> client for the <a href="https://nats.io">NATS messaging system</a>.
+</p>
+
+## Status
+[![License Apache 2](https://img.shields.io/badge/License-Apache2-blue.svg)](https://www.apache.org/licenses/LICENSE-2.0)
+[![Crates.io](https://img.shields.io/crates/v/nats.svg)](https://crates.io/crates/nats)
+[![Documentation](https://docs.rs/nats/badge.svg)](https://docs.rs/nats/)
+[![Build Status](https://travis-ci.org/nats-io/nats.rs.svg?branch=master)](http://travis-ci.org/nats-io/nats.rs)
+
+Warning: as of version 0.4.0, this crate has been rewritten, and several public
```
Comment (derekcollison) [inline CRATE.md:15]: Would say we are switching code bases to a Synadia (official maintainers) of NATS.io

File: CRATE.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+APIs have changed. Please see the [docs](https://docs.rs/nats), the [new
+repo](https://github.com/nats-io/nats.rs), and [the
+changelog](https://github.com/nats-io/nats.rs/blob/master/CHANGELOG.md) for
+details.
+
+## Motivation
+Rust may be the most interesting new language the NATS ecosystem has seen. We
```
Comment (derekcollison) [inline CRATE.md:22]: Feel free to rewrite this..

File: CRATE.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+and having subscriptions generate multiple styles of iterators, since iterators
+are a first class citizen in Rust. We also wanted to be aligned with the NATS
+philosophy of simple, secure, and fast! (The security is coming soon!)
+
+## Feedback
+
+This is a new client, and I and the team are new to Rust, so all feedback welcome!
+We encourage all folks in the NATS and Rust ecosystems to help us improve this library. Please open issues, submit PRs, etc.
+
+## Example Usage
```
Comment (derekcollison) [inline CRATE.md:36]: Any new ones we want to add?

File: CRATE.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+`> cargo run --example nats-box -- -h`
+
+Basic connections, and those with options. The compiler will force these to be correct.
+
+```rust
+let nc = nats::connect("localhost")?;
+
+let nc2 = nats::ConnectionOptions::new()
+    .with_name("My Rust NATS App")
+    .with_user_pass("derek", "s3cr3t!")
```
Comment (derekcollison) [inline CRATE.md:47]: Maybe credentials based example too?

File: CRATE.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  * [X] User JWTs (NATS 2.0)
+* [X] Reconnect logic
+* [X] TLS support
+* [ ] Direct async support
+* [X] Crates.io listing
+
+### Miscellaneous TODOs
+* [ ] Ping timer
+* [X] msg.respond
+* [ ] Drain mode
```
Comment (derekcollison) [inline CRATE.md:144]: This one important.

File: CRATE.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+## Sync vs Async
+
+The Rust ecosystem has a diverse set of options for async behaviors. This
+client library can be used somewhat effectively already with async runtimes
+such as async-std and tokio. Going forward we look to provide an async client.
+Publish today is mostly non-blocking, so largest API change would be around
+subscriptions being streams vs iterators by default. Also been researching
+sinks and whether or not they make sense. Would probably be a config feature
+for async wnd options for most runtimes like async-std and tokio.
```
Comment (stjepang) [inline CRATE.md:124]:
```suggestion
for async and options for most runtimes like async-std and tokio.
```

File: CRATE.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+  * [X] Token
+  * [X] User/Password
+  * [X] Nkeys
+  * [X] User JWTs (NATS 2.0)
+* [X] Reconnect logic
+* [X] TLS support
+* [ ] Direct async support
+* [X] Crates.io listing
+
+### Miscellaneous TODOs
```
Comment (stjepang) [inline CRATE.md:141]: Perhaps add "Headers" too?

File: CRATE.md (hunk 4 above)
Comment (spacejam) [reply CRATE.md:43]: added auth and TLS constructor examples, fixed some compiler errors with some of the expamples

Comment (derekcollison) [reply CRATE.md:43]: NGS/Creds example?

Comment (derekcollison) [reply CRATE.md:43]: Also the "I am new to Rust" probably should be removed since you and Stjepan are more representative of the project now.

**Pull request merged**